packages/langium/src/utils/ast-util.ts

```diff
@@ -177,4 +179,17 @@ export function findLocalReferences(targetNode: AstNode, lookup: AstNode): Strea
     };
     streamAllContents(lookup).forEach(process);
     return stream(refs);
+}
+
+export function findAllReferences(targetNode: AstNode, astNodePath: string, referenceLookUp: IndexManager): Stream<AstNodeReferenceDescription> {
```

Comment:
Utility functions that take an injectable service as argument are a smell; see comment above.